### PR TITLE
Expose buy bitcoin on SDK flutter

### PR DESCRIPTION
**Pull request by ademar111190**
Related: https://github.com/breez/c-breez/pull/556

### CODE REVIEW

**Comment by roeierez** [comment]
@ademar111190 remember we need also to add the API to the React Native code.

File: libs/sdk-react-native/src/index.tsx [hunk 1]
```diff
@@ -577,3 +581,8 @@ export const recommendedFees = async (): Promise<RecommendedFees> => {
     const response = await BreezSDK.recommendedFees()
     return response as RecommendedFees
 }
+
+export const buyBitcoin = async (provider: BuyBitcoinProvider): Promise<string> => {
+    const response = await BreezSDK.buyBitcoin(provider)
```
Comment (roeierez) [inline libs/sdk-react-native/src/index.tsx:586]: This call should be implemented in both Android and iOS native modules:
BreezSDKModule.kt
RNBreezSDK.swift

**Review by roeierez — CHANGES_REQUESTED** [review]

File: libs/sdk-react-native/src/index.tsx (hunk 1 above)
Comment (ademar111190) [reply libs/sdk-react-native/src/index.tsx:586]: Just added.

File: libs/sdk-react-native/ios/RNBreezSDK.swift [hunk 2]
```diff
@@ -487,4 +487,16 @@ class RNBreezSDK: RCTEventEmitter {
             reject(RNBreezSDK.TAG, "Error calling recommendedFees", err)
         }
     }
+
+    @objc(buyBitcoin:rejecter:)
+    func buyBitcoin(_ provider: String, _ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
```
Comment (dangeross) [inline libs/sdk-react-native/ios/RNBreezSDK.swift:492]: Should be for consistency:
```
    @objc(buyBitcoin:resolver:rejecter:)
    func buyBitcoin(_ provider: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
```

File: libs/sdk-react-native/src/index.tsx (hunk 1 above)
Comment (dangeross) [reply libs/sdk-react-native/src/index.tsx:586]: Needs this adding to the bottom of `ios/RNBreezSDK.m`
```
RCT_EXTERN_METHOD(
    buyBitcoin: (NSString*)provider
    resolver: (RCTPromiseResolveBlock)resolve
    rejecter: (RCTPromiseRejectBlock)reject
)
```

**Comment by ademar111190** [comment]
Thanks @dangeross I just added these, let me know if there are more things.

File: libs/sdk-react-native/example/App.js [hunk 3]
```diff
@@ -113,6 +113,9 @@ const App = () => {
 
                 const fiatRates = await fetchFiatRates()
                 addLine("fetchFiatRates", JSON.stringify(fiatRates))
+
+                const buyBitcoin = await buyBitcoin()
```
[the remark this answers is not in the document]
Comment (ademar111190) [reply libs/sdk-react-native/example/App.js]: You are right, thanks once again!

**Review by dangeross — APPROVED** [review]

**Review by roeierez — APPROVED** [review]

**Comment by ademar111190** [comment]
Thank you guys for the reviews and help 🙏

**Pull request merged**